### PR TITLE
[WIP] Update monorepo to Zoe v0.2.0

### PR DESCRIPTION
**Pull request by katelynsills**
This PR is intended to update all of the monorepo to use Zoe v0.2.0 and to record the steps to do so and test the update. 

Currently, agoric-cli is erroring but it's not clear it's based on the Zoe update. I'm trying to debug. 

Let's talk about this Wednesday during the meeting.

### CODE REVIEW

**Comment by katelynsills** [comment]
Here's the error message (which I get when following our [Getting Started instructions](https://agoric.com/documentation/getting-started/), unrelated to updating Zoe, as far as I can tell).

```sh
katelynsills@Katelyns-MBP demo % npx agoric start
initializing solo
solo [ 'init', 'solo', '--egresses=none' ]
Cannot copy package-lock.json: Error: ENOENT: no such file or directory, copyfile '/Users/katelynsills/code/temp/demo/.agservers/node_modules/@agoric/cosmic-swingset/package-lock.json' -> 'solo/html/package-lock.json'
    at Proxy.copyFileSync (fs.js:1790:3)
    at initBasedir (/Users/katelynsills/code/temp/demo/.agservers/node_modules/@agoric/cosmic-swingset/lib/ag-solo/init-basedir.js:61:8)
    at solo (/Users/katelynsills/code/temp/demo/.agservers/node_modules/@agoric/cosmic-swingset/lib/ag-solo/main.js:65:5)
    at Object.<anonymous> (/Users/katelynsills/code/temp/demo/.agservers/node_modules/@agoric/cosmic-swingset/bin/ag-solo:7:1)
    at Module._compile (internal/modules/cjs/loader.js:959:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:995:10)
    at Module.load (internal/modules/cjs/loader.js:815:32)
    at Function.Module._load (internal/modules/cjs/loader.js:727:14)
    at Function.Module.runMain (internal/modules/cjs/loader.js:1047:10)
    at internal/main/run_main_module.js:17:11 {
  errno: -2,
  syscall: 'copyfile',
  code: 'ENOENT',
  path: '/Users/katelynsills/code/temp/demo/.agservers/node_modules/@agoric/cosmic-swingset/package-lock.json',
  dest: 'solo/html/package-lock.json'
}
ag-solo initialized in solo
HTTP/WebSocket will listen on 127.0.0.1:8000
Run '(cd solo && /Users/katelynsills/code/temp/demo/.agservers/node_modules/@agoric/cosmic-swingset/bin/ag-solo start)' to start the vat machine
(node:60323) Warning: require() of ES modules is not supported.
require() of /Users/katelynsills/code/temp/demo/.agservers/node_modules/@agoric/bundle-source/src/index.js from /Users/katelynsills/code/temp/demo/.agservers/node_modules/@agoric/cosmic-swingset/lib/ag-solo/bundle.js is an ES module file as it is a .js file whose nearest parent package.json contains "type": "module" which defines all .js files in that package scope as ES modules.
Instead rename index.js to end in .cjs, change the requiring code to use import(), or remove "type": "module" from /Users/katelynsills/code/temp/demo/.agservers/node_modules/@agoric/bundle-source/package.json.
linking html directories
solo [ 'start', '--role=three_client' ]
adding HTTP/WS listener on 127.0.0.1:8000
Serving static files from /Users/katelynsills/code/temp/demo/.agservers/solo/html
Listening on 127.0.0.1:8000
= loading config from basedir /Users/katelynsills/code/temp/demo/.agservers/solo/vats
ignoring  bootstrap.js
ignoring  gci.js
ignoring  lib-exchange.js
ignoring  lib-wallet.js
ignoring  pubsub.js
ignoring  repl.js
ignoring  scratch.js
127.0.0.1:51556[1]: new ws connection /
kernel.addDevice(mailbox)
HTTP: POST /vat 200 34 - 31.233 ms
127.0.0.1:51556[1]: client closed
kernel.addDevice(command)
kernel.addDevice(timer)
= adding vat 'exchange' from /Users/katelynsills/code/temp/demo/.agservers/solo/vats/vat-exchange.js
= adding vat 'host' from /Users/katelynsills/code/temp/demo/.agservers/solo/vats/vat-host.js
= adding vat 'http' from /Users/katelynsills/code/temp/demo/.agservers/solo/vats/vat-http.js
= adding vat 'mints' from /Users/katelynsills/code/temp/demo/.agservers/solo/vats/vat-mints.js
= adding vat 'pixel' from /Users/katelynsills/code/temp/demo/.agservers/solo/vats/vat-pixel.js
= adding vat 'provisioning' from /Users/katelynsills/code/temp/demo/.agservers/solo/vats/vat-provisioning.js
= adding vat 'registrar' from /Users/katelynsills/code/temp/demo/.agservers/solo/vats/vat-registrar.js
= adding vat 'sharing' from /Users/katelynsills/code/temp/demo/.agservers/solo/vats/vat-sharing.js
= adding vat 'uploads' from /Users/katelynsills/code/temp/demo/.agservers/solo/vats/vat-uploads.js
= adding vat 'wallet' from /Users/katelynsills/code/temp/demo/.agservers/solo/vats/vat-wallet.js
= adding vat 'zoe' from /Users/katelynsills/code/temp/demo/.agservers/solo/vats/vat-zoe.js
= adding vat 'vattp' from /Users/katelynsills/code/temp/demo/.agservers/node_modules/@agoric/swingset-vat/src/vats/vat-tp/vattp.js
= adding vat 'comms' from /Users/katelynsills/code/temp/demo/.agservers/node_modules/@agoric/swingset-vat/src/vats/comms/index.js
= adding vat 'timer' from /Users/katelynsills/code/temp/demo/.agservers/node_modules/@agoric/swingset-vat/src/vats/vat-timerWrapper.js
= adding vat '_bootstrap' from /Users/katelynsills/code/temp/demo/.agservers/solo/vats/bootstrap.js
loading bootstrap.js
wasInitialized = false
Assigned VatID v1 for Genesis vat exchange
Assigned VatID v2 for Genesis vat host
Assigned VatID v3 for Genesis vat http
Assigned VatID v4 for Genesis vat mints
error running ag-solo: TypeError: hp.resolve is not a function
    at passStyleOf (/Users/katelynsills/code/temp/demo/.agservers/solo/vats/vat-mints.js:817:18)
    at mustBeSameStructureInternal (/Users/katelynsills/code/temp/demo/.agservers/solo/vats/vat-mints.js:1454:23)
    at mustBeSameStructure (/Users/katelynsills/code/temp/demo/.agservers/solo/vats/vat-mints.js:1531:5)
    at mustBeComparable (/Users/katelynsills/code/temp/demo/.agservers/solo/vats/vat-mints.js:1538:5)
    at makeUnitOps (/Users/katelynsills/code/temp/demo/.agservers/solo/vats/vat-mints.js:1838:5)
    at makeMint (/Users/katelynsills/code/temp/demo/.agservers/solo/vats/vat-mints.js:2167:21)
    at build (/Users/katelynsills/code/temp/demo/.agservers/solo/vats/vat-mints.js:2258:38)
    at eval (/Users/katelynsills/code/temp/demo/.agservers/solo/vats/vat-mints.js:2275:55)
    at build (eval at <anonymous> (eval at createScopedEvaluatorFactory (/Users/katelynsills/code/temp/demo/.agservers/node_modules/ses/dist/ses.cjs.js:1308:10)), <anonymous>:1553:24)
    at Object.makeLiveSlots (eval at <anonymous> (eval at createScopedEvaluatorFactory (/Users/katelynsills/code/temp/demo/.agservers/node_modules/ses/dist/ses.cjs.js:1308:10)), <anonymous>:1574:9)
Maybe the chain has bumped and you need to rerun ag-setup-solo?
```

**Comment by michaelfig** [comment]
This error is because the published npm packages are out of date.  Cutting new releases will probably fix it.

**Comment by katelynsills** [comment]
> This error is because the published npm packages are out of date. Cutting new releases will probably fix it.

I also used `yarn link` to try to bypass that. It's possible that I was unsuccessful at bypassing. Is there a better way to be able to test before cutting new releases?

**Comment by michaelfig** [comment]
> I also used `yarn link` to try to bypass that. It's possible that I was unsuccessful at bypassing. Is there a better way to be able to test before cutting new releases?

It's really hard to use `yarn link` in this circumstance due to the nested and generated `package.json`s.  We'll discuss this at today's meeting.

**Comment by katelynsills** [comment]
Sounds good! In the meeting, let's walk through installing the agoric-cli in its current state with the old version of Zoe as well. I'm having a hard time understanding how it is erroring seemingly unrelated to the update (so maybe I don't understand where and how it uses Zoe?)

File: packages/zoe/CONTRIBUTING.md [hunk 1]
```diff
@@ -46,4 +46,56 @@ Before submitting a pull request, please:
...
+should be updated in a PR reviewed by the owners of the packages.
+Those packages are:
+* packages/agoric-cli (`packages/agoric-cli/template/contract/package.json`)
+* packages/cosmic-swingset (`packages/cosmic-swingset/package.json`)
+
+To test that that the Zoe update works (in the most basic sense) with cosmic-swingset, do:
+
+1. Run `yarn install` from the workspace root
+2. Run `yarn build` from the workspace root
+3. `cd packages/cosmic-swingset`
```
Comment (michaelfig) [inline packages/zoe/CONTRIBUTING.md]: Arguably, you could just test `agoric-cli` as long as you use the `--sdk` option, and when you run `agoric --sdk start` that gives you a working scenario3.  Testing the deploy scripts can be a separate step.

But this is fine for now.  We can refine the steps once `agoric-cli` is fully working.

File: packages/zoe/CONTRIBUTING.md [hunk 2]
```diff
@@ -46,4 +46,56 @@ Before submitting a pull request, please:
...
+4. `rm -rf t3`
+4. `make scenario3-setup`
+5. `make scenario3-run-client`
+6. Open `http://127.0.0.1:8000/`
+8. `moolaAssayP = E(home.registrar).get("moola_2059");`
```
Comment (michaelfig) [inline packages/zoe/CONTRIBUTING.md]: These steps can be made into a script that is run with `agoric deploy`.  That allows you to script any kind of interaction using tildot and the running `ag-solo`.

File: packages/zoe/CONTRIBUTING.md [hunk 3]
```diff
@@ -46,4 +46,56 @@ Before submitting a pull request, please:
...
+    kind: 'waived' }});`
+18. `seatAndPayoutP = E(home.zoe).redeem(exclInviteP, offerRules,
+    harden([undefined, undefined]));`
+19. `seatAndPayoutP.then(obj => {seat = obj.seat; payoutP = obj.payout});`
+20. `E(seatAndPayout.seat).makeOffer();`
+21. `seatAndPayout.payout.then(p => moolaPayment = p[0]);`
+20. `E(moolaPayment).getBalance();` (should be an extent of 0)
+
+
+To test that the update works with agoric-cli, do:
```
Comment (michaelfig) [inline packages/zoe/CONTRIBUTING.md]: Please remove these steps, and maybe mark with a TODO that we still need to make them work.

**Review by michaelfig — APPROVED** [review]
LGTM as another good step forward.

Maybe you would like to finish putting the simple exchange contract into `template/contract`, but when you are satisfied, feel free to merge.

I would prefer to fix the other `agoric-cli` problems (`api`, `ui`) in separate PRs, so that we can land that incrementally.  Don't worry that it doesn't currently work... it was broken before Zoe 0.2.0 too.

File: packages/zoe/CONTRIBUTING.md (hunk 1 above)
Comment (katelynsills) [reply packages/zoe/CONTRIBUTING.md]: Good point

File: packages/zoe/CONTRIBUTING.md (hunk 2 above)
Comment (katelynsills) [reply packages/zoe/CONTRIBUTING.md]: Ah, great. I didn't realize we already had that capability. That's much better. I'll switch to that in another PR.

File: packages/zoe/CONTRIBUTING.md (hunk 3 above)
Comment (katelynsills) [reply packages/zoe/CONTRIBUTING.md]: I rewrote these steps to match the current scenario, with a TODO where we need progress.

**Pull request merged**